framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -390,30 +390,37 @@ public S initialStore(UnderlyingAST underlyingAST, List<LocalVariableNode> param
     // the declared type instead of a refined type. Issue a warning to alert users?
     private void addInitialFieldValues(S store, ClassTree classTree, MethodTree methodTree) {
         boolean isConstructor = TreeUtils.isConstructor(methodTree);
+        boolean isStaticMethod =
+                ElementUtils.isStatic(TreeUtils.elementFromDeclaration(methodTree));
         TypeElement classEle = TreeUtils.elementFromDeclaration(classTree);
         for (FieldInitialValue<V> fieldInitialValue : analysis.getFieldInitialValues()) {
             VariableElement varEle = fieldInitialValue.fieldDecl.getField();
+            boolean fieldNotFromEnclosingClass = varEle.getEnclosingElement().equals(classEle);
             // Insert the value from the initializer of private final fields.
             if (fieldInitialValue.initializer != null
                     // && varEle.getModifiers().contains(Modifier.PRIVATE)
                     && ElementUtils.isFinal(varEle)
                     && analysis.atypeFactory.isImmutable(ElementUtils.getType(varEle))) {
                 store.insertValue(fieldInitialValue.fieldDecl, fieldInitialValue.initializer);
+                continue; // This insert is more specific compared with the below two inserts.
```

Comment:
Add more explicitly what this means. Write the comment on the line before the `continue`.

Reply:
sure